src/transformState.js

```diff
@@ -5,7 +5,27 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(state, action.extraData);
```

Comment:
this can also be achieved using [spread operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax) which is more common nowadays

Reply:
I'm not sure about spread operator in this case. I tried, but tests were failed. So, I left the option with Object.assign().

Reply:
That is correct solution in this case, spread operator is great on it's own, but the task forces us to think differently